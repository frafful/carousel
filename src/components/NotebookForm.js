import React from 'react';
import moment from 'moment';

export default class NotebookForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: props.notebook ? props.notebook.name : '',
      createdAt: props.notebook ? moment(props.notebook.createdAt) : moment(),
      error: ''
    };
  }

  onSubmit = (e) => {
    e.preventDefault();

    if(!this.state.name) {
      this.setState(() => ({
        error: 'Favor informar o nome do caderno'
      }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        name: this.state.name,
        createdAt: this.state.createdAt.valueOf()
      });
    }
  };

  onNameChange = (e) => {
    const name = e.target.value;
    this.setState(() => ({name}));
  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input 
            type="text"
            placeholder="TÃ­tulo"
            value={this.state.name}
            onSubmit={this.onSubmit}
            onChange={this.onNameChange}
          />
          <button>Criar Caderno</button>
        </form>
      </div>
    );
  }
};