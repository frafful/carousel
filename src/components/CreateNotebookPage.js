import React from 'react';
import { connect } from 'react-redux';
import NotebookForm from './NotebookForm';
import { startAddNotebook } from '../actions/notebookList';

export class CreateNotebook extends React.Component { 
  
  onSubmit = (notebook) => {
    this.props.startAddNotebook(notebook);
    this.props.history.push('/');
  };
  
  render() {
    return (
      <div>
        <h1>Crie um novo Caderno</h1>
        <NotebookForm onSubmit={this.onSubmit} />
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => {
  return { 
    startAddNotebook: (notebook) => dispatch(startAddNotebook(notebook))
  };
};

export default connect(undefined, mapDispatchToProps)(CreateNotebook);