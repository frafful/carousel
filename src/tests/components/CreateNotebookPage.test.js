import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { CreateNotebook } from '../../components/CreateNotebookPage';
import notebooks from '../fixtures/notebooks';

test('should render create notebook page correctly', () => {
  const wrapper = shallow(<CreateNotebook />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  const startAddNotebook = jest.fn();
  const history = { push: jest.fn() };
  
  const wrapper = shallow(<CreateNotebook startAddNotebook={startAddNotebook} history={history} />);
  wrapper.find('NotebookForm').prop('onSubmit')(notebooks[1]);
  
  expect(startAddNotebook).toHaveBeenLastCalledWith(notebooks[1]);
  expect(history.push).toHaveBeenLastCalledWith('/');
});