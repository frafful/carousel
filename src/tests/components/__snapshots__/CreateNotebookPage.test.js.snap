// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render create notebook page correctly 1`] = `
<div>
  <h1>
    Crie um novo Caderno
  </h1>
  <NotebookForm
    onSubmit={[Function]}
  />
</div>
`;
